src/common/ceph_json.h

```diff
@@ -502,7 +502,9 @@ template<class T>
 static void encode_json_impl(const char *name, const T& val, ceph::Formatter *f)
 {
   f->open_object_section(name);
+#ifdef WITH_RADOSGW_RADOS
```

Comment:
This one we can't do .  We can't put RGW specific defines into libcommon.   We'll have to find the code that calls this and either fix it or ifdef it out.